src/Soil-Core/SoilIndexIterator.class.st

```diff
@@ -400,8 +400,15 @@ SoilIndexIterator >> removeItemsSuchThat: aBlock [
 
 { #category : #removing }
 SoilIndexIterator >> removeKey: key [
+	^ self 
+		removeKey: key 
+		ifAbsent: [ KeyNotFound signalFor: key in: self ]
+]
+
+{ #category : #removing }
+SoilIndexIterator >> removeKey: key ifAbsent: aBlock [
 	| returnValue |
-	returnValue := self at: key.
+	returnValue := self at: key ifAbsent: aBlock.
```

Comment:
Doing two times #at: is expensive.  We should have that picking up at the current position

Reply:
yes, we did this already, this pr moves the check now to the iterator (this at least avoids instantiating it twice)

But we have to do this in one step for sure, I will check how to do this

Reply:
I have implemented SoilIndexIterator>>#removeKey:ifAbsent: to search only once